packages/dai/src/eth/smartContract/wrapContract.js

```diff
@@ -34,8 +37,29 @@ export function wrapContract(contract, name, abi, txManager) {
         if (key in target) return target[key];
         if (!txManager || !nonConstantFns[key]) return contract[key];
 
-        return (...args) =>
-          txManager.sendContractCall(contract, key, args, name);
+        return (...args) => {
+          const lastArg = args[args.length - 1];
+
+          // If the last arg is a business object, don't count it as a function input
+          const functionInputsLength =
+            typeof lastArg === 'object' &&
+            lastArg !== null &&
+            lastArg.constructor === Object
+              ? args.length - 1
+              : args.length;
+
+          for (const fnKey in contract.interface.functions) {
+            // Match the function override with key that has the same number of inputs
+            if (
+              contract.interface.functions[fnKey].name === key &&
+              contract.interface.functions[fnKey].inputs.length ===
+                functionInputsLength
+            ) {
+              key = fnKey;
+            }
+          }
+          return txManager.sendContractCall(contract, key, args, name);
+        };
```

Comment:
We now have to be more specific with function overloads in ethers 5, so we have to specify the exact function signature, rather than the function name. Here we determine the correct key by matching the number of inputs from the interface with the number of args passed in (minus the business object at the end).